mllib/src/main/scala/org/apache/spark/mllib/util/MLUtils.scala

```diff
@@ -82,6 +82,8 @@ object MLUtils {
           val value = indexAndValue(1).toDouble
           (index, value)
         }.unzip
+        require(indices.size == 0 || indices(0) >= 0,
```

Comment:
Just another conditional? seems vanishingly small compared to other work done here. The current check assumes that it's in correct libsvm format and that the indices are ordered ascending. The code doesn't actually rely on this though and works even with missorted indices. However the current check wouldn't work if "0:..." occurred later in the line.